### ISSUE TITLE
Variable K8s Metric Frequencies

### ISSUE THREAD

**Issue by ikelockett** [splunk/splunk-connect-for-kubernetes#181]
- Increasing the frequency to 15s for all metrics is extremely resource inefficient. Some metrics are needed high frequency, while others we could do with having at a lower resolution.
- For instance, 'limit' metrics change very slowly (perhaps hourly, daily) and don't need to be monitored and onboarded at the same rate as 'usage' metrics which we might want at 1s-15s intervals
- Ideal solution would allow us to wildcard metric names and set custom frequencies for them.

E.g.

*.cpu.limit: 1h
*.cpu.usage: 15s
*.diskspace.usage: 5m

For an idea of impact, in order to get just 4-5 critical metrics at 15s frequency, we would have to also import 5000+ low-priority metrics at the same frequency, which isn't scalable.

**Comment by beth2016** [comment]
Is there an eta for this?  We really need this as well.  thank you.

**Comment by chaitanyaphalak** [comment]
@beth2016 @ikelockett whats the highest resolution that you need for this implementation, @ikelockett you have mentioned 1s-15s intervals, anything below 5s will index a substantially large amount of data in the metrics index. Let me know your thoughts. Thanks.

**Comment by ikelockett** [comment]
I'm sure it will differ per implementation, but 10s frequency would be a sensible default for the higher frequency metrics

**Comment by halr9000** [comment]
@jayntguru @alanzchan comments?

**Comment by jayntguru** [comment]
We found 15 seconds to be exceptionally large as-is. Right now we're at 60 seconds in a lab cluster  with 9 nodes including control plane/infra and a bunch of random things deployed. That's about 400mb/day. Our current "solution" (bash running an oc command that dumps to a file that's picked up by splunk) gathers these stats once a day so we're happy with the 60 second resolution and we're prepared for a longer time (than 60 seconds) in busier clusters.

What could be more clear as long as we're discussing this is exactly where to make the change. There are two config maps -

splunk-splunk-kubernetes-metrics
splunk-splunk-kubernetes-metrics-aggregator

I made a change to the one that isn't the aggregator and achieved my desired results. Not totally sure if that's the correct place to change it though.

-JS
